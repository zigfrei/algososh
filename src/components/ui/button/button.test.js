import renderer from "react-test-renderer";
import { Button } from "../button/button";
import {
  render,
  fireEvent,
} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

describe("testing Button component for render and Match Snapshot", () => {
  it("button with text", () => {
    const ButtonWithText = renderer.create(<Button text="Текст кнопки" />);
    expect(ButtonWithText).toMatchSnapshot();
  });
  it("button without text", () => {
    const ButtonWithoutText = renderer.create(<Button />);
    expect(ButtonWithoutText).toMatchSnapshot();
  });
  it("disabled button", () => {
    const DisabledButton = renderer.create(<Button disabled={true} />);
    expect(DisabledButton).toMatchSnapshot();
  });
  it("button loading", () => {
    const LoadingButton = renderer.create(<Button isLoader={true} />);
    expect(LoadingButton).toMatchSnapshot();
  });
});

describe("should calls the handler", () => {
  it("on click", () => {
    const handler = jest.fn();
    const { getByTestId } = render(
      <Button text="Кнопка для теста" onClick={handler} data-testid="button" />
    );
    const TestButton = getByTestId("button");
    //проверяем является ли кнопкой
    expect(TestButton.tagName).toBe("BUTTON");
    //нажимаем на кнопку
    fireEvent.click(TestButton);
    //проверяем вызвался ли callback
    expect(handler).toHaveBeenCalled();
  });
});