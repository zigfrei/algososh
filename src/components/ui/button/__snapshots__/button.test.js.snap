// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Button component for render and Match Snapshot button loading 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`testing Button component for render and Match Snapshot button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Текст кнопки
  </p>
</button>
`;

exports[`testing Button component for render and Match Snapshot button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`testing Button component for render and Match Snapshot disabled button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
